File: 234-palindrome-linked-list/palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        stack <int> stack;
        ListNode* cur = head;
        while(cur){
            stack.push(cur->val);
            cur = cur->next;
        }
        cur = head;
        while(cur && cur->val == stack.top()){
            stack.pop();
            cur = cur->next;
        }
        return cur == nullptr;
    }
};

File: 234-palindrome-linked-list/palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        cur = head
        list=[]
        while(cur):
            list.append(cur.val)
            cur = cur.next
        cur = head
        while(cur and cur.val == list.pop()):
            cur = cur.next
        return cur == None